lib/harvester_utils/downloader.rb

```diff
@@ -27,6 +33,19 @@ def self.download_file(url)
       file
     end
 
+    def self.extract_csv(zip_file)
+      Zip::File.open(zip_file) do |zip|
+        csv_file = zip.glob('*.csv')
+        raise NoCSVError, "No CSV file found in the provided zip_file: #{zip_file}" if csv_file.nil?
+
+        file = Tempfile.new(rand(1_000_000).to_s)
+        file.binmode
+        file.write(csv_file.first.get_input_stream.read)
+        file.close
+        file
```

Comment:
Tu nemusis robit random, to robi tempfile sam + odporucam block verziu, nemusis potom zatvarat. 

https://ruby-doc.org/stdlib-2.5.6/libdoc/tempfile/rdoc/Tempfile.html#method-c-open